newsrevenuehub/rev-engine
# DEV-1893 - DEV-1893_force_https

#### What's this PR do?

Asks Django to redirect any non-TLS requests (http) to HTTPS.

#### Why are we doing this? How does it help us?

Security.

#### How should this be manually tested?

```
python manage.py runserver
curl -I http://localhost:8000/
```
Confirm that you see something like:
```
HTTP/1.1 301 Moved Permanently
...
Location: https://localhost:8000/
```
```
SECURE_SSL_REDIRECT=False python manage.py runserver
curl -I http://localhost:8000/
```
Confirm that you get a 200:
```
HTTP/1.1 200 OK
...
```

#### Have automated unit tests been added?

No.

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

No.

#### Has this been documented? If so, where?

No.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-1893

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No, but for local development you'll probably want to set `SECURE_SSL_REDIRECT=False` unless you want to SSL locally.

#### Are there next steps? If so, what? Have tickets been opened for them?

No.
